### PR TITLE
feat: [CDS-59012]: BE changes to support Delegate Selection Logs in Service Infra Step

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
## Describe your changes
To support Delegate Selection Logs in Service Infra Step
Sending the delegate task id in step details  using SdkGraphVisualizationDataService::publishStepDetailInformation and UI can use that to fetch delegate selection logs
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46773)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java [hunk 1]
```diff
@@ -103,6 +103,7 @@
 public class ArtifactsStepV2 implements AsyncExecutableWithRbac<EmptyStepParameters> {
   private static final long DEFAULT_TIMEOUT = TimeUnit.MINUTES.toMillis(3);
   static final String ARTIFACTS_STEP_V_2 = "artifacts_step_v2";
+  private static final String ARTIFACT_TASK = "ARTIFACT_TASK";
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java:106]: lets rename this to something more meaningful like StepDetailKey etc

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java [hunk 2]
```diff
@@ -106,6 +106,7 @@ public class ConfigFilesStepV2 extends AbstractConfigFileStep
   private static final String CONFIG_FILES_STEP_V2 = "CONFIG_FILES_STEP_V2";
   static final String CONFIG_FILE_COMMAND_UNIT = "configFiles";
   static final int CONFIG_FILE_GIT_TASK_TIMEOUT = 10;
+  private static final String CONFIG_FILES_TASK = "CONFIG_FILES_TASK";
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java:109]: lets rename this to something more meaningful like StepDetailKey etc

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java [hunk 3]
```diff
@@ -220,12 +223,15 @@ public class CDStepHelper {
...
   @Inject private CDExpressionResolver cdExpressionResolver;
+  @Inject private SdkGraphVisualizationDataService sdkGraphVisualizationDataService;
+  @Inject private ServiceStepsHelper serviceStepsHelper;
 
   public static final String RELEASE_NAME_VALIDATION_REGEX =
       "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*";
 
   public static final Pattern releaseNamePattern = Pattern.compile(RELEASE_NAME_VALIDATION_REGEX);
   public static final String GIT = "/_git/";
+  public static final String TASK_IDS = "Task IDs";
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java:234]: lets rename this to something more meaningful like StepDetailTaskIdKey etc

File: 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/InfrastructureTaskExecutableStepV2.java [hunk 4]
```diff
@@ -124,6 +126,7 @@ public class InfrastructureTaskExecutableStepV2 extends AbstractInfrastructureTa
                                                .setType(ExecutionNodeType.INFRASTRUCTURE_TASKSTEP_V2.getName())
                                                .setStepCategory(StepCategory.STEP)
                                                .build();
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/InfrastructureTaskExecutableStepV2.java:128]: lets rename this to something more meaningful like StepDetailKey etc

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/beans/InfraStepV2DetailsInfo.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import io.harness.pms.sdk.core.steps.executables.StepDetailsInfo;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+@RecasterAlias("io.harness.cdng.infra.beans.InfraStepV2DetailsInfo")
+public class InfraStepV2DetailsInfo implements StepDetailsInfo {
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/beans/InfraStepV2DetailsInfo.java:22]: Lets jusgt call this InfrastructureStepDetails ( get rid of v2 etc )

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/ServiceStepV3DetailsInfo.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.Set;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+@RecasterAlias("io.harness.cdng.service.beans.ServiceStepV3DetailsInfo")
+public class ServiceStepV3DetailsInfo implements StepDetailsInfo {
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/ServiceStepV3DetailsInfo.java:23]: Lets jusgt call this ServiceStepDetails ( get rid of v3 etc )

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java (hunk 1 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java:106]: Done

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java (hunk 2 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java:109]: Done

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java (hunk 3 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java:234]: removed it, as its not required

File: 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/InfrastructureTaskExecutableStepV2.java (hunk 4 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/InfrastructureTaskExecutableStepV2.java:130]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/ServiceStepV3DetailsInfo.java (hunk 6 above)
Comment (alluvamsi-harness) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/service/beans/ServiceStepV3DetailsInfo.java]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/beans/InfraStepV2DetailsInfo.java (hunk 5 above)
Comment (alluvamsi-harness) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/beans/InfraStepV2DetailsInfo.java]: Dobne

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 4 files at r2, 5 of 6 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46773)* status: 9 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by alluvamsi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46773)* status: 9 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java` line 1061 at r4](https://reviewable.io/reviews/harness/harness-core/46773#-NTnBdxT5Fd3fKhUwjVL:-NTnbiJKEcE02qGLw_XO:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/43c63e249f9c9e2f0af2d3e5a7eedb7dd96c4da2/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java#L1061)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

better to keep this logic in ServiceStepHelper instead of CDStepHelper. this is not a generic CD utility.
</blockquote></details>

Done

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46773)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46773)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Comment by alluvamsi-harness** [comment]
trigger ng_manager

**Review by sainath-b — APPROVED** [review]

**Comment by alluvamsi-harness** [comment]
trigger review

**Pull request merged**